### PR TITLE
Added a new Maths algorithm to determine if two non-null integers are "friendly numbers"

### PR DESCRIPTION
**Pull request by fun-guava**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

Added a new FriendlyNumbers.js file, with a Maths algorith to determine if two integers are friendly numbers.  Friendaly number are defined as follows (source wikipedia):   'In number theory, friendly numbers are two or more natural numbers with a common abundancy index, the  ratio between the sum of divisors of a number and the number itself.'
This is my first ever real pull request as I am totally new to git and gitHub. Please don't hesitate to let me now if I am not following the proper procedure.

- [x ] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/JavaScript/pull/1267"></a>

### CODE REVIEW

File: Maths/FriendlyNumbers.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  // input: two integers
+  // output: true if the two integers are friendly numbers, false if they are not friendly numbers
+
+  // First, check that the parameters are valid
+  if (!Number.isInteger(firstNumber) || !Number.isInteger(secondNumber) || firstNumber === 0 || secondNumber === 0 || firstNumber === secondNumber) {
+    throw new Error('The two parameters must be distinct, non-null integers')
+  }
+
+  // Calculate the abundancy index of the two number.
+  // ... first get the sum of their divisors
```
Comment (appgurueu) [inline Maths/FriendlyNumbers.js:18]: could use a function for that

File: Maths/FriendlyNumbers.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+export const FriendlyNumbers = (firstNumber, secondNumber) => {
+  // input: two integers
+  // output: true if the two integers are friendly numbers, false if they are not friendly numbers
+
+  // First, check that the parameters are valid
+  if (!Number.isInteger(firstNumber) || !Number.isInteger(secondNumber) || firstNumber === 0 || secondNumber === 0 || firstNumber === secondNumber) {
+    throw new Error('The two parameters must be distinct, non-null integers')
+  }
+
+  // Calculate the abundancy index of the two number.
```
Comment (appgurueu) [inline Maths/FriendlyNumbers.js:17]: could write a function to do that

File: Maths/FriendlyNumbers.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  for (let i = 0; i < secondNumber / 2; i++) {
+    if (Number.isInteger(secondNumber / i)) {
+      sumDivisorsSecondNumber += i
+    }
+  }
+  // ... and divide that sum by the number itself
+  const abundancyIndexFirstNumber = sumDivisorsFirstNumber / firstNumber
+  const abundancyIndexSecondNumber = sumDivisorsSecondNumber / secondNumber
+
+  if (abundancyIndexFirstNumber === abundancyIndexSecondNumber) {
```
Comment (appgurueu) [inline Maths/FriendlyNumbers.js:36]: just `return abundancyIndex(firstNumber) == abundancyIndex(secondNumber);`, assuming you write a function `abundancyIndex`.

File: Maths/FriendlyNumbers.js (hunk 2 above)
Comment (fun-guava) [reply Maths/FriendlyNumbers.js]: Absolutely! Thanks for the advice. I commited the updated code. 
Please let me know if something is amiss (I'm completely new to gitHub and git....)

**Review by appgurueu — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**